eventhubs/client/azure-eventhubs/src/main/java/com/azure/eventhubs/EventReceiver.java

```diff
@@ -15,17 +24,81 @@
  * A {@link EventReceiver#receive()} is tied to a Event Hub PartitionId + consumer group combination.
  *
  * <ul>
- *      <li>If the {@link EventReceiver} is created where {@link EventReceiverOptions#exclusiveReceiverPriority()} has a
- *      value, then Event Hubs service will guarantee only 1 active receiver exists per partitionId and consumer group
- *      combination. This is the recommended approach to create a {@link EventReceiver}.</li>
- *      <li>Multiple receivers per partitionId and consumer group combination can be created using non-epoch receivers.</li>
+ * <li>If the {@link EventReceiver} is created where {@link EventReceiverOptions#exclusiveReceiverPriority()} has a
+ * value, then Event Hubs service will guarantee only 1 active receiver exists per partitionId and consumer group
+ * combination. This is the recommended approach to create a {@link EventReceiver}.</li>
+ * <li>Multiple receivers per partitionId and consumer group combination can be created using non-epoch receivers.</li>
  * </ul>
  *
  * @see EventHubClient#createReceiver(String)
  * @see EventHubClient#createReceiver(String, EventReceiverOptions)
  */
-public class EventReceiver {
-    private PartitionProperties partitionInformation = null;
+public class EventReceiver implements Closeable {
+    private static final AtomicReferenceFieldUpdater<EventReceiver, AmqpReceiveLink> RECEIVE_LINK_FIELD_UPDATER =
+        AtomicReferenceFieldUpdater.newUpdater(EventReceiver.class, AmqpReceiveLink.class, "receiveLink");
+
+    private final Mono<AmqpReceiveLink> receiveLinkMono;
+    private final Duration operationTimeout;
+    private final AtomicInteger creditsToRequest = new AtomicInteger(1);
+    private final AtomicBoolean isDisposed = new AtomicBoolean();
+    private final ServiceLogger logger = new ServiceLogger(EventReceiver.class);
+    private final EmitterProcessor<EventData> emitterProcessor;
+    private final Flux<EventData> messageFlux;
+
+    private volatile PartitionProperties partitionInformation = null;
+    private volatile AmqpReceiveLink receiveLink;
+
+    EventReceiver(Mono<AmqpReceiveLink> receiveLinkMono, EventReceiverOptions options, Duration operationTimeout) {
+        this.receiveLinkMono = receiveLinkMono;
+        this.emitterProcessor = EmitterProcessor.create(options.prefetchCount(), false);
+        this.operationTimeout = operationTimeout;
```

Comment:
I thought that .NET was the only platform that wasn't removing the Receiver Timeout thinger (we made it non-public).  Track one uses it, but I thought the general consensus was that it wasn't applicable because of the streaming receiver approach.

Reply:
This operation timeout is only used when we want to close the Receiver. I don't want to block indefinitely waiting for an instance of the `AmqpReceiveLink` when all I want to do is dispose of it.